bridge/src/app.rs

```diff
@@ -27,54 +26,93 @@ pub struct App<T> where T: Transport {
 
 pub struct Connections<T> where T: Transport {
 	pub home: T,
+	pub home_url: RpcUrlKind,
 	pub foreign: T,
+	pub foreign_url: RpcUrlKind,
 }
 
 impl Connections<Http>  {
-	pub fn new_http(handle: &Handle, home: &str, home_concurrent_connections: usize, foreign: &str, foreign_concurrent_connections: usize) -> Result<Self, Error> {
+	/// Returns new home and foreign HTTP transport connections, falling back
+	/// to failover urls if necessary.
+	pub fn new_http(handle: &Handle, home_primary: &RpcUrl, home_failover: Option<&RpcUrl>,
+			home_concurrent_connections: usize, foreign_primary: &RpcUrl,
+			foreign_failover: Option<&RpcUrl>, foreign_concurrent_connections: usize)
+			-> Result<Self, Error> {
+		// Attempts to connect to either a primary or failover url, returning
+		// the transport and the url upon success.
+		fn connect(handle: &Handle, url_primary: &RpcUrl, url_failover: Option<&RpcUrl>,
+				concurrent_connections: usize) -> Result<(Http, RpcUrlKind), Web3Error> {
+			match Http::with_event_loop(&url_primary.to_string(), handle, concurrent_connections) {
```

Comment:
This is the most important part of my review. My reading of https://github.com/tomusdrw/rust-web3/blob/master/src/transports/http.rs#L78 suggests that this call will never fail if primary is not available as it doesn't attempt the connection, meaning bridge will always try primary, then fail, restart and try it again to no avail.

Would love to be proven wrong, though -- if I misread any part or something.

Reply:
I think you're right which is why I wanted to make clear that this PR doesn't properly address the issue. It looks like the connection errors are passed up through the spawned future. I'll need to dig around and look into how those are currently handled (see below).